### PR TITLE
Add DuchyIdConfig for internal Kingdom services

### PR DESCRIPTION
**Pull request by uakyol**
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)

### CODE REVIEW

**Comment by uakyol** [comment]
Had a hard time with naming but DuchyIdConfig is the best I could come up with. Happy to consider other options

**Review by wangyaopw — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @oliver-amzn, @SanjayVas, @uakyol, and @zachcwc)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 21 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf580p-0ArGbNgc7r7E:-Mf56dLo-I8jRKSbnsG1:b-xrip1i) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d27ecf0cb678f55850aa57170e0b5e6c2df03e12/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L21)):*
> ```Kotlin
> import picocli.CommandLine
> 
> object DuchyIds {
> ```

this object only needs to support two functions, getInternalId(externalId: String): Long and getExternalId(internal: Long): String.

It  can contain two maps for easy querying (ExternalToInternalIdMap, InternalToExternalIdMap), or just a List of pairs (quering by filtering the pairs)

I feel it doesn't need count ALL_DUCHY_EXTERNAL_IDS, getByDuchyExternalId, Entry etc.

**Review by uakyol — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @oliver-amzn, @SanjayVas, @wangyaopw, and @zachcwc)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 21 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf580p-0ArGbNgc7r7E:-Mf9aC4K0xMAP7-QSg26:bcvghu3) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d27ecf0cb678f55850aa57170e0b5e6c2df03e12/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L21)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

this object only needs to support two functions, getInternalId(externalId: String): Long and getExternalId(internal: Long): String.

It  can contain two maps for easy querying (ExternalToInternalIdMap, InternalToExternalIdMap), or just a List of pairs (quering by filtering the pairs)

I feel it doesn't need count ALL_DUCHY_EXTERNAL_IDS, getByDuchyExternalId, Entry etc.
</blockquote></details>

simplified it, thanks

**Review by wangyaopw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 0 of 9 files reviewed, 4 unresolved discussions (waiting on @oliver-amzn, @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

___
*[src/main/k8s/duchy_id_config.textproto, line 2 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9aPNJAv-m10CxD2z4:-Mf9aPNKEJjl3vULUEQd:b-a2olkw) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/k8s/duchy_id_config.textproto#L2)):*
> ```
> DuchyIdonfig
> ```
typo
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 24 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bJwG1KdSOMtU09CG:-Mf9bJwG1KdSOMtU09CH:b1u9nvo) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L24)):*
> ```
>   val count: Int
>     get() = DuchyIds.entries.size
> ```
delete if not useful?
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 41 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bRca7V4hqPflvR1M:-Mf9bRca7V4hqPflvR1N:b-kmd4cc) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L41)):*
> ```
> entries.firstOrNull { it.internalDuchyId == internalDuchyId }?.externalDuchyId
> ```
since the array size is small, likely only 2 or 3, doing filtering should be fine.
But I personally feel using 2 maps are more efficient. No strong opinion any way

**Review by uakyol — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 0 of 9 files reviewed, 4 unresolved discussions (waiting on @oliver-amzn, @SanjayVas, @wangyaopw, and @zachcwc)

___
*[src/main/k8s/duchy_id_config.textproto, line 2 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9aPNJAv-m10CxD2z4:-Mf9ijTY0tPmMEojVIYv:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/k8s/duchy_id_config.textproto#L2)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

> ```
> DuchyIdonfig
> ```
typo
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 24 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bJwG1KdSOMtU09CG:-Mf9iiTC0AofNR3B0cNP:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L24)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

> ```
>   val count: Int
>     get() = DuchyIds.entries.size
> ```
delete if not useful?
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 41 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bRca7V4hqPflvR1M:-Mf9iUFE88A2H9xl0Qty:bsf9cf3) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L41)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

> ```
> entries.firstOrNull { it.internalDuchyId == internalDuchyId }?.externalDuchyId
> ```
since the array size is small, likely only 2 or 3, doing filtering should be fine.
But I personally feel using 2 maps are more efficient. No strong opinion any way
</blockquote></details>

yeah, I added a comment on why we are iterating.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 6 of 9 files at r1, 1 of 1 files at r2, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 7 unresolved discussions (waiting on @oliver-amzn, @uakyol, @wangyaopw, and @zachcwc)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 41 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bRca7V4hqPflvR1M:-MfAS60mDXNLjrgYSZIt:bf7h7ze) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L41)):*
<details><summary><i>Previously, uakyol wrote…</i></summary><blockquote>

yeah, I added a comment on why we are iterating.
</blockquote></details>

1. I would adjust the comment and move to a second paragraph for clarity. e.g. something like "Note that this performs an O(n) linear scan, where *n* is the number of Duchies. This should be sufficient as the number of Duchies is expected to be small (~5)."
2. Perhaps consider a one-way map at least? That way at least one direction gets easier lookups and the other can do a linear scan on values.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 1 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfARJ5OCpje9UQ45fuo:-MfARJ5OCpje9UQ45fup:bcrysvw) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/bc6321bfc95318942fa344cec8c45cae3e98156b/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L1)):*
> ```Kotlin
> // Copyright 2021 The Cross-Media Measurement Authors
> ```

Move this file somewhere in the kingdom tree since nothing else should ever use it. (We actually want to eventually remove all references to Duchy/Kingdom from `common` such that there's nothing in this repo's `common` and everything comes from common-jvm).
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 22 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfATbIqE223Kruunvsz:-MfATbIqE223Kruunvt-:bul2zyv) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/bc6321bfc95318942fa344cec8c45cae3e98156b/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L22)):*
> ```
> lateinit var
> ```
This should be private.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 29 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfAUHb33CExYRn1LOOP:-MfAUHb33CExYRn1LOOQ:b-pkwoxp) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/bc6321bfc95318942fa344cec8c45cae3e98156b/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L29)):*
> ```
> toTypedArray()
> ```
Array isn't a great collection type to use. Usually it's only when you happen to be interacting with something lower-level. Why do this at all and not just keep the List returned by `map`?

**Review by uakyol — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 8 of 9 files reviewed, 7 unresolved discussions (waiting on @oliver-amzn, @SanjayVas, @wangyaopw, and @zachcwc)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 41 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bRca7V4hqPflvR1M:-MfFAZP_D7d5ENswZeKB:bp248q9) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L41)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

1. I would adjust the comment and move to a second paragraph for clarity. e.g. something like "Note that this performs an O(n) linear scan, where *n* is the number of Duchies. This should be sufficient as the number of Duchies is expected to be small (~5)."
2. Perhaps consider a one-way map at least? That way at least one direction gets easier lookups and the other can do a linear scan on values.
</blockquote></details>

1. Thank you, sounds much cleaner
2. If  you don't mind I want to keep it this way since there is no computational value in doing this and the code would become a little more complex. But if you feel strongly, let me know and I will change.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 1 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfARJ5OCpje9UQ45fuo:-MfF9F1ADq_tkBBF4pIW:b-9b6fbd) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/bc6321bfc95318942fa344cec8c45cae3e98156b/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L1)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Move this file somewhere in the kingdom tree since nothing else should ever use it. (We actually want to eventually remove all references to Duchy/Kingdom from `common` such that there's nothing in this repo's `common` and everything comes from common-jvm).
</blockquote></details>

moved up one folder based on discussion
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 22 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfATbIqE223Kruunvsz:-MfFANp63_OwWXgMtfMn:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/bc6321bfc95318942fa344cec8c45cae3e98156b/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L22)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

> ```
> lateinit var
> ```
This should be private.
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 29 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfAUHb33CExYRn1LOOP:-MfF9zFpEhSxQuXLTVEb:b-fjtymj) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/bc6321bfc95318942fa344cec8c45cae3e98156b/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L29)):*
> lateinit var
basically was trying to match what was implemented before: https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/f82b5b780cd52af78e4195109184a7055ccb9cab/src/main/kotlin/org/wfanet/measurement/common/identity/DuchyInfo.kt#L21  but I agree, changed it.

**Review by zachcwc — APPROVED** [review]
> Reviewed 6 of 9 files at r1, 1 of 1 files at r2, 1 of 2 files at r4, 5 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 7 unresolved discussions (waiting on @oliver-amzn, @SanjayVas, and @wangyaopw)

**Review by SanjayVas — APPROVED** [review]
> Reviewed 4 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 5 unresolved discussions (waiting on @oliver-amzn, @uakyol, and @wangyaopw)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt, line 33 at r5](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfKMkfMBg5JFcaq_G7x:-MfKMkfMBg5JFcaq_G7y:bfkekyg) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5bbf3d2a4ed738191ff01860b18ae48a9bfc36b7/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt#L33)):*
> ```
> Note
> ```
nit: Make this a second paragraph. In KDoc, this means leaving a blank line between it and the summary.

See https://developer.android.com/kotlin/style-guide#paragraphs
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 41 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bRca7V4hqPflvR1M:-MfKN4MZAkWl7aMC-QJT:b-r9sz8) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L41)):*
<details><summary><i>Previously, uakyol wrote…</i></summary><blockquote>

1. Thank you, sounds much cleaner
2. If  you don't mind I want to keep it this way since there is no computational value in doing this and the code would become a little more complex. But if you feel strongly, let me know and I will change.
</blockquote></details>

That's fine.

**Review by oliver-amzn — COMMENTED** [review]
Why add a new config flag/object/etc and scatter duchy-related information across several places when we have `DuchyInfo.kt` which we could extend? 

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-Mfcd_YM9Q6G68K-A4pS:b-y7csu2)* status: 9 of 10 files reviewed, 5 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

**Review by SanjayVas — COMMENTED** [review]
DuchyInfo is intended to be used by both Kingdom and Duchy for RPC info. This is for external to internal ID mapping, which is specific to the Kingdom internal services and shouldn't be exposed elsewhere.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-MfchOUM2rrHePz60thl:b-devoo7)* status: 9 of 10 files reviewed, 5 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

**Review by oliver-amzn — COMMENTED** [review]
Why define it so narrowly to "just RPC info" and not have all duchy-related metadata in there? 

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-Mfd3cZ00fG7lhLEEJz0:b-upjyeg)* status: 9 of 10 files reviewed, 5 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

**Review by oliver-amzn — COMMENTED** [review]
My concern is that we keep adding more and more boilerplate code and fragment what arguably is all duchy-related information. 

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-Mfd3oJr8trwZ2foOGKv:b-k31o89)* status: 9 of 10 files reviewed, 5 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

**Review by SanjayVas — COMMENTED** [review]
> Why define it so narrowly to "just RPC info" and not have all duchy-related metadata in there?

On purpose to exclude Duchy internal IDs, which even as a concept shouldn't be known outside of Kingdom internal services. This is specifically why I had requested adding "RPC" in the config proto file.

`DuchyInfo` is defined in a common location that can be shared by Kingdom and Duchy code. `DuchyIds` is defined in the `kingdom` package tree to ensure it's not e.g. referenced by Duchy code.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-Mfd7CE5-6kHvQgQ7FrE:b80p0zu)* status: 9 of 10 files reviewed, 5 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

**Review by oliver-amzn — COMMENTED** [review]
Hmmmkay.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-MfdMjuF34HC1zu2j6u2:b-rbf)* status: 9 of 10 files reviewed, 5 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

**Review by oliver-amzn — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 9 of 10 files reviewed, 6 unresolved discussions (waiting on @SanjayVas, @uakyol, @wangyaopw, and @zachcwc)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt, line 28 at r6](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfdN4a73vHx-Aw6MLwc:-MfdN4a84GT2OKGmqGQd:b-s3dti2) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/7d20d4d54b143b125728972b9e38135d3400a8d6/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt#L28)):*
> ```
> Duchy config
> ```

nit: --> `Duchy Id config` ?

**Review by uakyol — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: 7 of 11 files reviewed, 6 unresolved discussions (waiting on @oliver-amzn, @SanjayVas, @wangyaopw, and @zachcwc)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt, line 41 at r3](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-Mf9bRca7V4hqPflvR1M:-MfdaJ_f-mSI9YXn9fKO:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/3267446e7cbf27394e463d98244191db2cf17482/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/identity/DuchyIds.kt#L41)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

That's fine.
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt, line 33 at r5](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfKMkfMBg5JFcaq_G7x:-MfdkRbw3F2y_-HSs4H9:b-k73z76) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/5bbf3d2a4ed738191ff01860b18ae48a9bfc36b7/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt#L33)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

> ```
> Note
> ```
nit: Make this a second paragraph. In KDoc, this means leaving a blank line between it and the summary.

See https://developer.android.com/kotlin/style-guide#paragraphs
</blockquote></details>

thanks
___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt, line 28 at r6](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-MfdN4a73vHx-Aw6MLwc:-MfdjjVT6I9FYUFCo4Fu:b-thazcz) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/7d20d4d54b143b125728972b9e38135d3400a8d6/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/common/DuchyIds.kt#L28)):*
<details><summary><i>Previously, oliver-amzn (Oliver) wrote…</i></summary><blockquote>

> ```
> Duchy config
> ```

nit: --> `Duchy Id config` ? 
</blockquote></details>

ah thanks

**Review by oliver-amzn — APPROVED** [review]
> Reviewed 2 of 10 files at r1, 1 of 1 files at r2, 1 of 3 files at r4, 3 of 6 files at r5, 1 of 1 files at r6, 3 of 3 files at r7.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 6 unresolved discussions (waiting on @oliver-amzn, @SanjayVas, and @wangyaopw)

**Review by SanjayVas — APPROVED** [review]
> Reviewed 1 of 1 files at r6, 1 of 3 files at r7.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 5 unresolved discussions (waiting on @oliver-amzn and @wangyaopw)

**Review by zachcwc — APPROVED** [review]
> Reviewed 1 of 1 files at r6, 1 of 3 files at r7.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 5 unresolved discussions (waiting on @oliver-amzn and @wangyaopw)

**Review by oliver-amzn — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 4 unresolved discussions (waiting on @wangyaopw)

**Review by wangyaopw — COMMENTED** [review]
> Reviewed 2 of 10 files at r1, 1 of 1 files at r2, 1 of 3 files at r4, 3 of 6 files at r5, 1 of 1 files at r6, 1 of 3 files at r7.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 4 unresolved discussions (waiting on @wangyaopw)

**Review by zachcwc — APPROVED** [review]
> Reviewed 1 of 6 files at r8, 6 of 6 files at r9.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 4 unresolved discussions (waiting on @wangyaopw)

**Review by uakyol — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160#-:-MfkVAh6-Iyk2D0QShTF:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @wangyaopw)

**Review by wangyaopw — APPROVED** [review]
> Reviewed 2 of 3 files at r7, 1 of 6 files at r8, 6 of 6 files at r9.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: all files reviewed, 4 unresolved discussions (waiting on @wangyaopw)

**Review by wangyaopw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @uakyol)

**Review by wangyaopw — APPROVED** [review]
> Reviewed 2 of 2 files at r10.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/160)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @uakyol)
